### ISSUE TITLE
different field types

### ISSUE THREAD

**Issue by superstructor** [mamund/hal-forms#6]
Hi @mamund,

Thanks for the interesting spec! 

Have you considered how to represent different field types; e.g.
- date picker
- time picker 
- checkboxes 
- radio buttons
- toggles / switches
- select boxes
- value ranges
- file uploads
- email (on mobile uses different keyboard etc) 
- passwords
- lists / multi-selection select boxes
  etc

**Comment by mamund** [comment]
@superstructor 

good to see your feedback.

i've thought a *bit* about the rendering instructions, but not acted on them yet.

right now, the focus is on describing the arguments and their constraints (required, readOnly, regex).

adding rendering details is definitely a possibility and i'd like to see them as *optional* elements that appear as extensions of the `property` element.

```
{"name":"hatSize",
 "value":"",
 "prompt":"Select Hat Size",
 "required":true,
 "render": {
    "type":"dropdown",
    "options":[
      {"value":"7","text":"Regular"},
      {"value":"7.5","text":"Irregular"}
    ]
  }
}
````

much of this could be "stolen" from HTML5 but considerations must be made for native apps and other platforms, too.

as long as these are done as OPTIONAL extensions and designed in a way to make it easy to find/parse (and ignore) them, i'm cool with it.

if you're interested in doing a sample design (and associated implementation), feel free to do a fork and PR. love to see it.

cheers.

**Comment by mamund** [comment]
@superstructor 

note issue #9 addresses some of your options.

also, would be happy to see you design/suggest some of the features you have in your list as HAL-FORMS extensions

**Comment by mamund** [comment]
closed -- follow-ups in #36 and others.
